### DISCUSSION TITLE
result would exceed 2^31-1 bytes

### DISCUSSION THREAD

**Discussion by Xiuying** [posit-dev/positron#4983] [Q&A]
When using Positron version 2024.10.0.14 (R version 4.4.0) to read an .rds file with the command seurat_annot <- readDS("seurat_annot_celltype_id.rds"), I encounter the following error: "Error in paste0(lines, collapse = "\n"): result would exceed 2^31-1 bytes". However, the same file loads successfully in R and RStudio. What could be the reason for this, and how can I resolve the issue?

**Comment by DavisVaughan** [comment]
In ark we use `paste0(lines, collapse = "\n")` 3 times
https://github.com/search?q=repo%3Aposit-dev%2Fark+%22paste0%28lines%2C+collapse+%3D+%22&type=code

I imagine the culprit is likely this one related to tracebacks
https://github.com/posit-dev/ark/blob/f2fa6353f0077741b9b2846b9211eba2b59f32b5/crates/harp/src/modules/init.R#L62

My guess is something else went wrong when loading that RDS file and its trying to show you an error, but for some reason the traceback is absolutely massive (are we inlining something again?) and its causing this error.

- Is it possible for you to upload this RDS file for us to try?
- Can you please copy and paste everything you see in the Output channel for `R x.y.z: Kernel` after triggering the error?

**Reply by Xiuying** [reply]
First of all, thank you very much for your response and assistance. This is the result of a single-cell analysis using Seurat: An object of class Seurat with 28,204 features across 70,947 samples within 1 assay. Active assay: RNA (28,204 features, 2,000 variable features). Four dimensional reductions calculated: PCA, Harmony, UMAP, t-SNE. However, the file is quite large, making it inconvenient to upload. Interestingly, there are no issues when reading it separately using R or RStudio.

**Reply by juliasilge** [reply]
If you are able to create a smaller `.rds` that generates the problem, that would be super helpful for us!

You can find more details on how to get to the Output channel Davis mentioned [here](https://github.com/posit-dev/positron/wiki/Troubleshooting), but briefly, you would click on the Output tab and then choose the channel for the R kernel:

![Screenshot 2024-10-11 at 10 58 26 AM](https://github.com/user-attachments/assets/a39d6d80-92c5-4312-a135-be036f2c82aa)
